### DISCUSSION TITLE
How to useContractRead for an array of token id?

### DISCUSSION THREAD

**Discussion by wengyhong** [wevm/wagmi#1161] [Q&A]
Hello all,

I am trying to display a collection of NFTs that a wallet address owns from a particular contract address. 

I have a contract function 'tokensOfOwner' that take in owner address and returns an array of token id.

I have a contract function 'tokenURI' that takes in a single tokenId and return a string of tokenUri

How do I use useContractRead to call 'tokenURI' for each of the token id in the array?

I took a look in https://0.6.x.wagmi.sh/docs/hooks/useContractInfiniteReads#paginated-indexes , it's something similiar but I wanted to pass in a custom array instead of having fixed incremental value.

Any idea how to execute this using wagmi?

Thanks

**Comment by thekesslereffect** [comment]
Would like to know this as well. I can't seem to pass an array from useContractRead into the args of useContractWrite.

This is as far as I could get with obtaining the array from useContractRead. 


```js
const [tokenIdsOfOwner, setTokenIdsOfOwner] = useState([]);

const {data : wagmi_v3StakingTokensOfOwner } = useContractRead({
    addressOrName: '0x9963bc674e7D9c5259d5E0Bf1094d8D353f5dAA1',
    contractInterface: v3StakingContractABI,
    functionName: 'tokensOfOwner',
    args: [address],
    watch: true //update automatically
  })
  
function v3Staking_GetTokensOfOwner(){
    const res = wagmi_v3StakingTokensOfOwner;
    res.forEach(element => {
      tokenIdsOfOwner.push(element.toNumber())
    });
    setTokenIdsOfOwner(_tokensOfOwner);
    console.log(tokenIdsOfOwner);
    return tokenIdsOfOwner);
  }
```


But I cant seem to pass that array into useContractWrite for some reason


```js
// unstake
  const {config: wagmi_v3StakingUnstakeConfig} = usePrepareContractWrite({
    addressOrName: '0x9963bc674e7D9c5259d5E0Bf1094d8D353f5dAA1',
    contractInterface: v3StakingContractABI,
    functionName: 'unstake',
    args: [tokenIdsOfOwner],
  }) ;
  const { write: wagmi_v3StakingUnstake, isSuccess: wagmi_v3StakingUnstakeSuccess, } = useContractWrite(wagmi_v3StakingUnstakeConfig);
  
  function v3Staking_UnStake(){
    v3Staking_GetTokensOfOwner();
    wagmi_v3StakingUnstake?.();
  }
```
